config/config_params.go

```diff
@@ -240,12 +248,30 @@ func (config *Config) UpdateFrom(rawData map[string]string, source Source) (chan
 	return
 }
 
-func (c *Config) InterfacePrefixes() []string {
-	return strings.Split(c.InterfacePrefix, ",")
+func (config *Config) InterfacePrefixes() []string {
+	return strings.Split(config.InterfacePrefix, ",")
 }
 
-func (c *Config) InterfaceExcludes() []string {
-	return strings.Split(c.InterfaceExclude, ",")
+// InterfaceExcludes returns a list of Regexp objects, each representing either an
+// exact interface value match or a class of interface matches. This function will
+// panic if any individual value cannot be parsed / compiled into a Regexp.
+func (config *Config) InterfaceExcludes() []*regexp.Regexp {
+	log.Debugf("Compiling regexp values for InterfaceExclude %s \n", config.InterfaceExclude)
+	strValues := strings.Split(config.InterfaceExclude, ",")
+	regexpValues := make([]*regexp.Regexp, len(strValues))
+
+	for i, strValue := range strValues {
+		// All values in interface exclude list will be treated as regular
+		// expressions; any value not in regexp format will be converted into
+		// equivalent regexp that tests for exact match
+		if !RegexpIfaceElemRegexp.Match([]byte(strValue)) {
+			convertedValue := fmt.Sprintf("^%s$", strValue)
```

Comment:
As you [called out in an earlier thread](https://github.com/projectcalico/felix/pull/1980#discussion_r265684587), the previous validation for this field restricted characters to be from `[a-zA-Z0-9_-]` so it's probably safe without `QuoteMeta`. 

However, I kind of like the idea of using it, since it just completely rules out any possibility of wrong behaviour. I'll add `QuoteMeta`.

Reply:
Fixed.